src/community/leiden.c

```diff
@@ -1263,3 +1271,239 @@ igraph_error_t igraph_community_leiden(
 
     return IGRAPH_SUCCESS;
 }
+
+/**
+ * \function igraph_community_leiden_simple
+ * \brief Finding community structure using the Leiden algorithm, simple interface.
+ *
+ * This is a simplified interface to \ref igraph_community_leiden() for
+ * convenience purposes. Instead of requiring vertex weights, it allows
+ * choosing from a set of objective functions to maximize, and calculates
+ * the vertex weights accordingly. It also rescales the resolution parameter
+ * appropriately.
+ *
+ * \param graph The input graph. May be directed or undirected.
+ * \param weights The edge weights. If \c NULL, all weights are assumed to be 1.
+ * \param objective The objective function to maximize.
+ *    \clist
+ *    \cli IGRAPH_LEIDEN_OBJECTIVE_MODULARITY
+ *    Use the generalized modularity, defined as
+ *    <code>Q = 1/(2m) sum_ij (A_ij - γ k_i k_j / (2m)) δ(c_i, c_j)</code>
+ *    for undirected graphs and as
+ *    <code>Q = 1/m sum_ij (A_ij - γ k^out_i k^in_j / m) δ(c_i, c_j)</code>
+ *    for directed graphs. This effectively uses a multigraph configuration
+ *    model as the null model. Edge weights must not be negative.
+ *    \cli IGRAPH_LEIDEN_OBJECTIVE_CPM
+ *    Use the constant Potts model, whose objective function is defined as
+ *    <code>Q = 1/(2m) sum_ij (A_ij - γ) δ(c_i, c_j)</code>
+ *    for undirected graphs and as
+ *    <code>Q = 1/m sum_ij (A_ij - γ) δ(c_i, c_j)</code>
+ *    for directed graphs. Edge weights are allowed to be negative.
+ *    \cli IGRAPH_LEIDEN_OBJECTIVE_ER
+ *    Use an objective function based on the Erdős-Rényi G(n,p) null model,
+ *    defined as
+ *    <code>Q = 1/(2m) sum_ij (A_ij - γ p) δ(c_i, c_j)</code>
+ *    for undirected graphs and as
+ *    <code>Q = 1/m sum_ij (A_ij - γ) δ(c_i, c_j)</code>
+ *    for directed graphs. \c p is the weighted density, i.e. the average
+ *    link strength between all vertex pairs (whether adjacent or not).
+ *    It is computed with the assumption that self-loops cannot be present
+ *    in the graph. If self-loops are present, an error is raised.
+ *    Edge weights must not be negative.
+ *    \cli IGRAPH_LEIDEN_OBJECTIVE_ER_LOOPS
+ *    Same as \c IGRAPH_LEIDEN_OBJECTIVE_ER but computes the density with
+ *    the assumption that self-loops may be present in the graph.
+ *    \endclist
+ *    In the above formulas, \c A is the adjacency matrix, \c m is the total
+ *    edge weight, \c k are the (out- and in-) degrees, \c γ is the resolution
+ *    parameter, and <code>δ(c_i, c_j)</code> is 1 if vertices \c i and \c j
+ *    are in the same community and 0 otherwise.
+ * \param resolution The resolution parameter, which is represented by γ in
+ *    the objective functions detailed above.
+ * \param beta The randomness used in the refinement step when merging. A small
+ *    amount of randomness (\c beta = 0.01) typically works well.
```

Comment:
Can we use `beta = 0.01` in `functions.yaml` as a reasonable default for `beta`? I'm trying to provide most of the arguments with a default value if this is indeed meant as a simpler alternative to the full Leiden function.

Reply:
It's already there, but we also need to add a default for `start`, i.e. `start=False`. I'll do that.